paasta_tools/tron_tools.py

```diff
@@ -231,6 +233,16 @@ def get_spark_paasta_pool(self):
     def get_env(self):
         env = super().get_env()
         spark_env = {}
+        if "aws_credentials_yaml" in self.config_dict:
+            access_key, secret_key = get_aws_credentials(
+                service=self.get_service(),
+                aws_credentials_yaml=self.config_dict.get("aws_credentials_yaml"),
+            )
+            env["AWS_ACCESS_KEY_ID"] = access_key
+            env["AWS_SECRET_ACCESS_KEY"] = secret_key
+            env["AWS_DEFAULT_REGION"] = self.config_dict.get(
```

Comment:
Can you confirm that we actually need this?
Every configuration option that we add is an opportunity or developers to get something wrong.
Many aws client libraries auto-detect their region, do we need to be explicit here and ask the user?

Reply:
In the current implementation we do allow user to provide both `--aws-credentials-yaml` and `--aws-region` options to `paasta spark-run` in their tron action definition. You may search for `--aws-region` in soa config codebase and see actual use case of those 2 options.
This bug fix aims to migrate aws-related options that user uses in their current tron action config from spark-run options to action config fields. So IMO these config options are needed as they are actively being used.

Reply:
For AWS region, I really don't think we need the extra option. If the job needs to use a non-default value (which no one is right now, every `--aws-region` option for spark is just us-west-2 in soaconfigs), we already have a way to configure env variables. They can add AWS_DEFAULT_REGION to the env section of the action, e.g.

```
my_spark_action:
  executor: spark
  env:
    AWS_DEFAULT_REGION: us-east-1
```

And then in this function, we can set AWS_DEFAULT_REGION to us-west-2 if it hasn't already been set by the user.

Reply:
I agree with @qui about the env var, we don't need a separate option for this.  I'm wondering if we should default to `us-west-2` or "whatever region the master is running in", though?

Reply:
Let's make it `us-west-2` for now at least. That's the behavior of paasta spark-run.